primitives/core/src/crypto.rs

```diff
@@ -106,8 +106,16 @@ impl<T: Zeroize> std::fmt::Debug for Protected<T> {
 	}
 }
 
-impl<T: Zeroize> From<T> for Protected<T> {
-	fn from(t: T) -> Self {
+#[cfg(feature = "std")]
+impl From<String> for Protected<String> {
```

Comment:
Can we not just open this `From<T: Zeroize>`?

Reply:
The problem is with type as [u8; N] where zeroize is implemented but for whom Protected would be moving/copying stack memory all around. Typically String and Vec are relatively safe to use (as long as we don't reallocate), otherwhise Box<T> would be safe but the api change a bit for it.
Then it does not change a thing with existing code, it just prevents some misuses.

Reply:
`Protected` does not implement `Copy`.

Next point is that you can not implement `Drop` for `Copy`able types: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ffa7796c63f95857904cdce1e0a3ebc5

Reply:
I was more refering to moves or memory copy of the stack that result from compiling. But the PR just move this case out to avoid such discussion on the public api.
Another way to handle this could be to add a comment.